src/main/java/com/codeborne/selenide/impl/Navigator.java

```diff
@@ -39,6 +43,19 @@ public void open(URL url, String domain, String login, String password) {
     navigateToAbsoluteUrl(url.toExternalForm());
   }
 
+  public void open(String relativeOrAbsoluteUrl, AuthenticationType authenticationType, String login, String password) {
+    getAndCheckWebDriver();
+    Credentials credentials = new Credentials(login, password);
+    String authorization = String.format("%s %s", authenticationType.getValue(), credentials.encode());
+    getSelenideProxy().addRequestFilter("headers.request", (request, contents, messageInfo) -> {
+      final HttpHeaders headers = request.headers();
+      headers.add("Authorization", authorization);
+      headers.add("Proxy-Authorization", authorization);
```

Comment:
@rosolko Isn't it a problem that we add this filter, but never remove?
It means that it will add `Authorization` header to ALL subsequent requests. Is it ok? Or we should remove this filter?

Reply:
@asolntsev 
1. Previously you sad that we shouldn't call internal `getProxy()` and call only top level selenide proxy `getSelenideProxy()`
2. `autoAuthorization` method supports only `BASIC` auth type
3. If we remove filter isn't application logout user after another action in application?
4. If remove - how can we remove filters from selenide proxy? I see only add functionality.

Reply:
1. Agree
2. Agree
3. I don't know exactly if logout happens. Probably. But anyway, we need to be able to somehow disable filter (when user calls `open` without basicauth) or reconfigure it (when user calls `open` with another username/password).
4. Right, it seems that we cannot remove filter from BMP. So, we can add filter only once, and enable/disable or configure existing filter.